### ISSUE TITLE
darwin-rebuild: Incorrect usage description

### ISSUE THREAD

**Issue by PhilipBorgesen** [LnL7/nix-darwin#812]
Hey there, new user of nix-darwin. I like it so far but ran into some trouble using `darwin-rebuild`.

[`darwin-rebuild`][darwin-rebuild] reports the following usage description:
```
darwin-rebuild [--help] {edit | switch | activate | build | check | changelog}
               [--list-generations] [{--profile-name | -p} name] [--rollback]
               [{--switch-generation | -G} generation] [--verbose...] [-v...]
               [-Q] [{--max-jobs | -j} number] [--cores number] [--dry-run]
               [--keep-going] [-k] [--keep-failed] [-K] [--fallback] [--show-trace]
               [-I path] [--option name value] [--arg name value] [--argstr name value]
               [--flake flake] [--update-input input flake] [--impure] [--recreate-lock-file]
               [--no-update-lock-file] [--refresh] ...
```
The `--update-input input flake` option [as implemented][implemented] only takes one value however (`input`) and requires the `--flake` option to also be passed. The following usage description (`[--flake flake [--update-input input]]`) would be more correct and less confusing:
```
darwin-rebuild [--help] {edit | switch | activate | build | check | changelog}
               [--list-generations] [{--profile-name | -p} name] [--rollback]
               [{--switch-generation | -G} generation] [--verbose...] [-v...]
               [-Q] [{--max-jobs | -j} number] [--cores number] [--dry-run]
               [--keep-going] [-k] [--keep-failed] [-K] [--fallback] [--show-trace]
               [-I path] [--option name value] [--arg name value] [--argstr name value]
               [--flake flake [--update-input input]] [--impure] [--recreate-lock-file]
               [--no-update-lock-file] [--refresh] ...
```
I haven't checked whether other options depend on the `--flake` option.

[darwin-rebuild]: https://github.com/LnL7/nix-darwin/blob/master/pkgs/nix-tools/darwin-rebuild.sh#L14
[implemented]: https://github.com/LnL7/nix-darwin/blob/master/pkgs/nix-tools/darwin-rebuild.sh#L74-L76

**Comment by Enzime** [comment]
Could you make a PR for this? I think all the options that appear after `--flake` are for flakes, I think `--update-input input flake` should actually be `--override-input input flake` and `--update-input input` should be a separate flag

**Comment by PhilipBorgesen** [comment]
@Enzime: Here's the PR as requested: https://github.com/LnL7/nix-darwin/pull/1115

**Comment by Enzime** [comment]
Closed as #1115 as been merged
